pkg/image/build.go

```diff
@@ -156,6 +167,38 @@ func Build(cfg *config.Config, dir, imageName string, secrets []string, noCache,
 		global.LabelNamespace + "has_init": "true",
 	}
 
+	if isUsingCogBaseImage {
+		labels[global.LabelNamespace + "using_cog_base_image"] = "true"
+		labels[global.LabelNamespace + "cog-base-image-name"] = cogBaseImageName
+
+		// get the last layer of the cog base image so that when we look at the built cog image,
+		// we know where the base image ends
+		// pull the base image, as we'll need to pull it anyway to build the cog image
+		// Q: are we ok with the cog base image being pulled to the user's machine? can fix this later by implementing 
+		// a manifest inspect which doesn't require a pull
+
+		err := docker.Pull(cogBaseImageName)
+		if err != nil {
+			return fmt.Errorf("Failed to pull cog base image: %w", err)
+		}
```

Comment:
Could we avoid pulling by using something like [go-containerregistry](https://github.com/google/go-containerregistry/blob/main/README.md) to inspect the manifest directly?

Reply:
The fetcher or inspect methods that are part of `crane` there looks promising - let me look into using that to prevent a base image pull.

Reply:
Added a TODO - since this requires a more more involved change involving getting hooked up with creds to use an http client to call the remote registry, will do this in a follow-up PR in the interest of getting this merged.

Reply:
We can use the `remote` package to get a new instance of an Image, and then get the manifest via `img.Manifest()`